### PR TITLE
Feature/match orders

### PR DESCRIPTION
**Pull request by jeremyjams**

### CODE REVIEW

File: src/itest/java/com/iexec/blockchain/IntegrationTests.java [hunk 1]
```diff
@@ -38,26 +78,201 @@ public void getMetrics() {
...
+        String workerpool = iexecHubService.createWorkerpool(buildRandomName("pool"),
+                30, 1);
+        System.out.println("Created workerpool: " + workerpool);
+        String datasetAddress = iexecHubService.createDataset(buildRandomName("data"),
+                "https://abc.com/def.jpeg",
+                BytesUtils.EMPTY_HEXASTRING_64,
+                30, 1);
+        System.out.println("Created datasetAddress: " + datasetAddress);
+
+        OrderSigner signer = new OrderSigner(chainConfig.getChainId(),
```
Comment (zguesmi) [inline src/itest/java/com/iexec/blockchain/IntegrationTests.java:98]: What about a common `OrderSignerService` to be injected whenever needed. We can keep that for a next PR.

File: src/itest/java/com/iexec/blockchain/IntegrationTests.java [hunk 2]
```diff
@@ -38,26 +78,201 @@ public void getMetrics() {
...
+                chainConfig.getHubAddress(),
+                credentialsService.getCredentials().getEcKeyPair());
+
+        AppOrder signAppOrder = signer.signAppOrder(buildAppOrder(appAddress));
+        WorkerpoolOrder signWorkerpoolOrder = signer.signWorkerpoolOrder(buildWorkerpoolOrder(workerpool));
+        DatasetOrder signDatasetOrder = signer.signDatasetOrder(buildDatasetOrder(datasetAddress));
+        RequestOrder signRequestOrder = signer.signRequestOrder(buildRequestOrder(signAppOrder,
```
Comment (zguesmi) [inline src/itest/java/com/iexec/blockchain/IntegrationTests.java:105]:
```suggestion
        AppOrder signedAppOrder = signer.signAppOrder(buildAppOrder(appAddress));
        WorkerpoolOrder signedWorkerpoolOrder = signer.signWorkerpoolOrder(buildWorkerpoolOrder(workerpool));
        DatasetOrder signedDatasetOrder = signer.signDatasetOrder(buildDatasetOrder(datasetAddress));
        RequestOrder signedRequestOrder = signer.signRequestOrder(buildRequestOrder(signAppOrder,
```

File: src/main/java/com/iexec/blockchain/broker/BrokerService.java [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+                    new RestTemplate().postForEntity(chainConfig.getBrokerUrl()
+                            + "/orders/match", brokerOrder, FillOrdersCliOutput.class);
+            if (responseEntity.getStatusCode().is2xxSuccessful()
+                    && responseEntity.getBody() != null) {
+                FillOrdersCliOutput dealResponse = responseEntity.getBody();
+                log.info("Matched orders [chainDealId:{}, tx:{}]", dealResponse.getDealid(), dealResponse.getTxHash());
+                return Optional.of(dealResponse.getDealid());
+            }
+        } catch (Throwable e) {
+            log.error("Failed to requestFinalize [requester:{}, app:{}, " +
```
Comment (zguesmi) [inline src/main/java/com/iexec/blockchain/broker/BrokerService.java:108]:
```suggestion
            log.error("Failed to request match order [requester:{}, app:{}, " +
```

**Review by zguesmi — APPROVED** [review]

File: src/itest/java/com/iexec/blockchain/IntegrationTests.java (hunk 2 above)
Comment (jeremyjams) [reply src/itest/java/com/iexec/blockchain/IntegrationTests.java]: Updated

File: src/itest/java/com/iexec/blockchain/IntegrationTests.java (hunk 1 above)
Comment (jeremyjams) [reply src/itest/java/com/iexec/blockchain/IntegrationTests.java]: Just created the OrderSignerService for you, but IMO we don't need it for now

**Pull request merged**